emspy/query/analyticset.py

```diff
@@ -23,109 +28,150 @@ def __init__(self, conn, ems_id, exclude_trees=[], exclude_patterns=[]):
             connection object
         ems_id: int
             EMS system id
-        exclude_trees: list
-            list of trees to exclude from the group update, such as:
-            "Engine Services", "ADI", "EMS Library"
-        exclude_patterns: list
-            list of regex patterns for exclusion of groups, such as:
-            r"Engine Services:\d{9}.*" to exclude all SSO subdirectories in Engine Services
+        analytic_set_path: str
+            The path to the analytic set
         """
         Asset.__init__(self, conn, "AnalyticSet")
         self._ems_id = ems_id
-        self._analytic_sets = []
-        self._analytics = []
-        self.exclude_trees = exclude_trees
-        self.exclude_patterns = exclude_patterns
-        if len(self.exclude_patterns) > 0:
-            self.exclude_patterns = [re.compile(pattern) for pattern in self.exclude_patterns]
-        self.update_list()
-
-    def __get_analytic_set_group(self, groupId):
-        if groupId in self.exclude_trees:
-            print('-- Excluding analytic set group {0}'.format(groupId))
+        self._analytic_set_path = analytic_set_path
+        self.__parse_path()
+
+    def __parse_path(self, path_only=False):
+        if not self._analytic_set_path:
+            self._group_id = None
+            self._analytic_set_id = None
+            self._analytic_set_name = None
+            self._analytic_set_description = None
             return
-        for pattern in self.exclude_patterns:
-            if re.match(pattern, groupId) is not None:
-                print('-- Excluding analytic set group {0}'.format(groupId))
-                return
-        try:
-            print('-- Fetching analytic set group {0}'.format(groupId))
-            _, dict_data = self._conn.request(
-                uri_keys=('analyticSet', 'analytic_set_group'),
-                uri_args=(self._ems_id, groupId)
-            )
-            self._analytic_sets.append(dict_data)
-            for group in dict_data['groups']:
-                self.__get_analytic_set_group(group['groupId'])
-        except urllib.error.HTTPError:
-            print('-- Failed to fetch analytic set group {0}'.format(groupId))
-
-    def update_list(self):
+        else:
+            # Apparently there is no easy why to escape all "\" special characters in python
+            # So have to do a find replace first
+            ref_dict = {
```

Comment:
I don't feel super strongly about this but it might be nice to support `/` as well as `\`

Reply:
Added support for both \ and /